spec/index.bs

```diff
@@ -4183,41 +4189,113 @@ has a default layout created and used instead.
 
 ### <dfn dictionary>GPUProgrammableStage</dfn> ### {#GPUProgrammableStage}
 
+A {{GPUProgrammableStage}} describes the entry point in the user-provided
+{{GPUShaderModule}} that controls one of the programmable stages of a [=pipeline=].
+
 <script type=idl>
 dictionary GPUProgrammableStage {
     required GPUShaderModule module;
     required USVString entryPoint;
+    record<USVString, double> constants;
 };
 </script>
 
-A {{GPUProgrammableStage}} describes the entry point in the user-provided
-{{GPUShaderModule}} that controls one of the programmable stages of a [=pipeline=].
+<dl dfn-for=GPUProgrammableStage dfn-type=dict-member>
+    : <dfn>constants</dfn>
+    ::
+        Specifies the values of [=pipeline-overridable=] constants which are [=statically accessed=]
+        by the entry point {{GPUProgrammableStage/entryPoint}} in the shader module
+        {{GPUProgrammableStage/module}}.
+
+        Each key-value pair must uniquely identify a [=pipeline-overridable=] constant
+        (by a string representing the numeric ID of the constant, if specified;
+        otherwise, by the constant's identifier name).
+
+        When the pipeline is executed, that constant will be set to the specified value
+        ([=converted to an IDL value|converted=] from `double` to
+            {{boolean}}, {{float}}, {{long}}, or {{unsigned long}},
+            depending on the [=wgsl-type=] of the [=wgsl-declaration=] in WGSL).
+
+        <div class=example>
+            WGSL code:
+
+            <pre highlight=rust>
+                [[override(0)]]    let has_point_light: bool = true; // Algorithmic control.
+                [[override(1200)]] let specular_param: f32 = 2.3;    // Numeric control.
+                [[override(1300)]] let gain: f32;                    // Must be overridden.
+                [[override]]       let width: f32 = 0.0;             // Specifed at the API level
+                                                                     //   using the name "width".
+                [[override]]       let depth: f32;                   // Specifed at the API level
+                                                                     //   using the name "depth".
+                                                                     //   Must be overridden.
+            </pre>
 
-<div algorithm>
-    <dfn abstract-op>validating GPUProgrammableStage</dfn>(stage, descriptor, layout)
-        **Arguments:**
-            - {{GPUShaderStage}} |stage|
-            - {{GPUProgrammableStage}} |descriptor|
-            - {{GPUPipelineLayout}} |layout|
+            JavaScript code, keeping defaults where possible:
 
-        Return `true` if all of the following conditions are satisfied:
+            <pre highlight=js>
+                {
+                    // ...
+                    constants: {
+                        1300: 2.0,  // "gain"
+                        depth: -1,  // "depth"
```

Comment:
```suggestion
                        depth: -1,  // "depth"
                        width: 1,   // "width"
```
Suggest adding one that has a default in the shader, so that the example doesn't accidentally imply that you can only provide constants for values without a default.

Reply:
Nevermind. The diff screwed up my reading comprehension and I didn't see the second example or the caption for this one.